tools/experiments/experiments.js

```diff
@@ -268,41 +281,59 @@ function updateExperimentRow(experiment) {
  * @return {boolean}
  */
 function isExperimentOn_(id) {
+  // TODO(#25205): remove this once the CDN stops supporting the AMP_CANARY
+  // cookie.
+  const optInCookieValue =
+    getCookie(window, '__Host-AMP_OPT_IN') || getCookie(window, 'AMP_CANARY');
   switch (id) {
     case EXPERIMENTAL_CHANNEL_ID:
-      return getCookie(window, 'AMP_CANARY') == AMP_CANARY_COOKIE.CANARY;
+      return [
+        AMP_OPT_IN_COOKIE.EXPERIMENTAL,
+        _LEGACY_AMP_OPT_IN_COOKIE.EXPERIMENTAL,
+      ].includes(optInCookieValue);
     case BETA_CHANNEL_ID:
-      return getCookie(window, 'AMP_CANARY') == AMP_CANARY_COOKIE.RC;
+      return [AMP_OPT_IN_COOKIE.BETA, _LEGACY_AMP_OPT_IN_COOKIE.BETA].includes(
+        optInCookieValue
+      );
     case RTV_CHANNEL_ID:
-      return RTV_PATTERN.test(getCookie(window, 'AMP_CANARY'));
+      return RTV_PATTERN.test(optInCookieValue);
     default:
       return isExperimentOn(window, /*OK*/ id);
   }
 }
 
 /**
- * Opts in to / out of the "canary" or "rc" runtime types by setting the
- * AMP_CANARY cookie.
- * @param {string} cookieState One of AMP_CANARY_COOKIE.{DISABLED|CANARY|RC} or
- *   a 15-digit RTV.
+ * Opts in to / out of the "beta" or "experimental" runtime types by setting the
+ * __Host-AMP_OPT_IN cookie.
+ * @param {string} cookieState One of the AMP_OPT_IN_COOKIE enum values, or a
+ *   15-digit RTV.
  */
 function setAmpCanaryCookie_(cookieState) {
   let validUntil = 0;
   if (RTV_PATTERN.test(cookieState)) {
     validUntil = Date.now() + RTV_COOKIE_MAX_AGE_MS;
-  } else if (cookieState != AMP_CANARY_COOKIE.DISABLED) {
+  } else if (cookieState != AMP_OPT_IN_COOKIE.DISABLED) {
     validUntil = Date.now() + COOKIE_MAX_AGE_MS;
   }
   const cookieOptions = {
-    // Set explicit domain, so the cookie gets sent to sub domains.
-    domain: location.hostname,
     allowOnProxyOrigin: true,
     // Make sure the cookie is available for the script loads coming from
     // other domains. Chrome's default of LAX would otherwise prevent it
     // from being sent.
     sameSite: SameSite.NONE,
     secure: true,
   };
+  setCookie(
+    window,
+    '__Host-AMP_OPT_IN',
+    cookieState,
+    validUntil,
+    cookieOptions
+  );
+  // TODO(#25205): remove this once the CDN stops supporting the AMP_CANARY
+  // cookie.
+  // Set explicit domain, so the cookie gets sent to sub domains.
+  cookieOptions.domain = location.hostname;
   setCookie(window, 'AMP_CANARY', cookieState, validUntil, cookieOptions);
```

Comment:
Two questions here.

1. The call to `setCookie` currently doesn't include `highestAvailableDomain` or `domain` in the options. Do they need to be used in light of the new restrictions?

https://github.com/ampproject/amphtml/blob/1c5d54b5e3e4a2c108a558fa66b9cc22ca61058e/src/cookies.js#L84-L125

2. Since `trySetCookie` does not throw if cookie setting fails, is there a risk of one cooking getting set but not the other? Would it make sense to verify that both were set and if not, unset them? (I suspect not, but thought it's worth mentioning.)

https://github.com/ampproject/amphtml/blob/1c5d54b5e3e4a2c108a558fa66b9cc22ca61058e/src/cookies.js#L225-L229

Reply:
`__Host-` prefixed cookies refuse to be set if they have the `Domain` value, this is by design: https://googlechrome.github.io/samples/cookie-prefixes/